mite_selenium/__init__.py

```diff
@@ -48,45 +52,145 @@ def start(self):
+                logger.warning(
+                    f"Timings may be inaccurate as protocol is not http/1.1: {protocol}"
+                )
+            metrics = {
+                "dns_lookup_time": timings["domainLookupEnd"]
+                - timings["domainLookupStart"],
+                "dom_interactive": timings["domInteractive"],
+                "js_onload_time": timings["domContentLoadedEventEnd"]
+                - timings["domContentLoadedEventStart"],
+                "page_weight": timings["transferSize"],
+                "render_time": timings["domInteractive"] - timings["responseEnd"],
+                "tcp_time": self._get_tcp_timing(timings),
+                "time_to_first_byte": timings["responseStart"] - timings["connectEnd"],
+                "time_to_interactive": timings["domInteractive"]
+                - timings["requestStart"],
+                "time_to_last_byte": timings["responseEnd"] - timings["connectEnd"],
+                "tls_time": self._get_tls_timing(timings),
+                "total_time": timings["duration"],
+            }
+            self._context.send(
+                "http_selenium_page_load_metrics",
+                **self._extract_and_convert_metrics_to_seconds(metrics),
+            )
+
+    def _extract_first_entry(self, entries):
+        if len(entries) != 1:
+            logger.error(
+                f"Performance entries did not return the expected count: expected 1 - actual {len(entries)}"
+            )
+            return
+        else:
+            return entries[0]
+
+    def _extract_and_convert_metrics_to_seconds(self, metrics):
+        converted_metrics = dict()
+        non_time_based_metrics = ["page_weight"]
+        for k, v in metrics.items():
+            if k not in non_time_based_metrics:
+                converted_metrics[k] = self._convert_ms_to_seconds(v)
+            else:
+                converted_metrics[k] = v
+        return converted_metrics
+
+    def _convert_ms_to_seconds(self, value_ms):
+        try:
+            return value_ms / 1000
+        except Exception:
+            return value_ms
+
+    def _retrieve_javascript_metrics(self):
+        try:
+            return self._remote.execute_script(
+                "return performance.getEntriesByType('resource')"
+            )
+        except Exception:
+            logger.error("Failed to retrieve resource performance entries")
+            return []
+
+    def _clear_resource_timings(self):
+        self._remote.execute_script("performance.clearResourceTimings()")
+
+    def get(self, url):
+        self._remote.get(url)
+        self._send_page_load_metrics()
+
+    def get_js_metrics_context(self):
+        return JsMetricsContext(self)
+
+    def wait_for_element(self, locator, timeout=5):
+        try:
+            return WebDriverWait(self._remote, timeout).until(
+                EC.presence_of_element_located(locator)
+            )
+        except Exception as te:
```

Comment:
I think you need to catch something more specific than just `Exception` here, otherwise, if there is some other sort of error (process interrupted?) then the logs will misleadingly say that there was a timeout

Reply:
You're right, this needs to be a timeout exception hence the `te`, not sure why I left it as Exception